test/python/result/test_result.py

```diff
@@ -731,3 +731,19 @@ def test_memory_int_out(self):
         self.assertEqual(
             'Result for experiment "99" could not be found.', context.exception.message
         )
+
+
+# This class tests all the same things as TestResultOperations, but removes the "creg_sizes"
+# instance-variable on experiment headers, so that we can check that when results are returned
+# from pulse backends, they are properly handled (since those results don't have creg_sizes
+# fields).
+
+
+class TestResultOperationsNoCreg(TestResultOperations):
+    """Result operations methods, but without creg_sizes."""
+
+    def generate_qiskit_result(self):
+        """Generate Result (without creg_sizes for testing"""
+        rv = super().generate_qiskit_result()
+        del rv.results[0].header.creg_sizes
+        return rv
```

Comment:
I think this test is a bit artificial on one side, and covers too much on the other one.

Reply:
Here is my suggestion, closer to the original issue.

```suggestion
    def test_marginal_counts_no_cregs(self):
        """Test that marginal_counts without cregs See qiskit-terra/6430."""
        raw_counts_1 = {"0x0": 4, "0x1": 7, "0x2": 10, "0x6": 5, "0x9": 11, "0xD": 9, "0x12": 8}
        data_1 = models.ExperimentResultData(counts=dict(**raw_counts_1))
        exp_result_header_1 = QobjExperimentHeader(memory_slots=5)
        exp_result_1 = models.ExperimentResult(
            shots=54, success=True, data=data_1, header=exp_result_header_1
        )

        result = Result(results=[exp_result_1], **self.base_result_args)

        marginal_counts_result = marginal_counts(result, indices=[0])
```